alacritty_terminal/src/term/mod.rs

```diff
@@ -1632,7 +1929,10 @@ impl<T: EventListener> Handler for Term<T> {
             ansi::Mode::LineFeedNewLine => self.mode.remove(TermMode::LINE_FEED_NEW_LINE),
             ansi::Mode::Origin => self.mode.remove(TermMode::ORIGIN),
             ansi::Mode::ColumnMode => self.deccolm(),
-            ansi::Mode::Insert => self.mode.remove(TermMode::INSERT),
+            ansi::Mode::Insert => {
+                self.mode.remove(TermMode::INSERT);
+                self.is_fully_damaged = true;
+            },
```

Comment:
Ah, this seems to relate to my comment in `fn damage()`.

I suppose your strategy is to reduce the number of times `fully_damaged` has to be set by only calling it in remove and checking `insert` in `fn damage()` basically? Since one or the other always has to be hit.

Reply:
Yeah, you could look at `INSERT` test, since it's covered in it.